schematypes.go

```diff
@@ -1488,102 +1509,296 @@ type DisassembledInstruction struct {
-func (*ContinueResponse) isMessage()                {}
-func (*NextRequest) isMessage()                     {}
-func (*NextResponse) isMessage()                    {}
-func (*StepInRequest) isMessage()                   {}
-func (*StepInResponse) isMessage()                  {}
-func (*StepOutRequest) isMessage()                  {}
-func (*StepOutResponse) isMessage()                 {}
-func (*StepBackRequest) isMessage()                 {}
-func (*StepBackResponse) isMessage()                {}
-func (*ReverseContinueRequest) isMessage()          {}
-func (*ReverseContinueResponse) isMessage()         {}
-func (*RestartFrameRequest) isMessage()             {}
-func (*RestartFrameResponse) isMessage()            {}
-func (*GotoRequest) isMessage()                     {}
-func (*GotoResponse) isMessage()                    {}
-func (*PauseRequest) isMessage()                    {}
-func (*PauseResponse) isMessage()                   {}
-func (*StackTraceRequest) isMessage()               {}
-func (*StackTraceResponse) isMessage()              {}
-func (*ScopesRequest) isMessage()                   {}
-func (*ScopesResponse) isMessage()                  {}
-func (*VariablesRequest) isMessage()                {}
-func (*VariablesResponse) isMessage()               {}
-func (*SetVariableRequest) isMessage()              {}
-func (*SetVariableResponse) isMessage()             {}
-func (*SourceRequest) isMessage()                   {}
-func (*SourceResponse) isMessage()                  {}
-func (*ThreadsRequest) isMessage()                  {}
-func (*ThreadsResponse) isMessage()                 {}
-func (*TerminateThreadsRequest) isMessage()         {}
-func (*TerminateThreadsResponse) isMessage()        {}
-func (*ModulesRequest) isMessage()                  {}
-func (*ModulesResponse) isMessage()                 {}
-func (*LoadedSourcesRequest) isMessage()            {}
-func (*LoadedSourcesResponse) isMessage()           {}
-func (*EvaluateRequest) isMessage()                 {}
-func (*EvaluateResponse) isMessage()                {}
-func (*SetExpressionRequest) isMessage()            {}
-func (*SetExpressionResponse) isMessage()           {}
-func (*StepInTargetsRequest) isMessage()            {}
-func (*StepInTargetsResponse) isMessage()           {}
-func (*GotoTargetsRequest) isMessage()              {}
-func (*GotoTargetsResponse) isMessage()             {}
-func (*CompletionsRequest) isMessage()              {}
-func (*CompletionsResponse) isMessage()             {}
-func (*ExceptionInfoRequest) isMessage()            {}
-func (*ExceptionInfoResponse) isMessage()           {}
-func (*ReadMemoryRequest) isMessage()               {}
-func (*ReadMemoryResponse) isMessage()              {}
-func (*DisassembleRequest) isMessage()              {}
-func (*DisassembleResponse) isMessage()             {}
+func (m *Request) GetSeq() int          { return m.Seq }
+func (m *Request) GetType() MessageType { return RequestType }
```

Comment:
It's a bit odd that Type is a string ("request", "response", "event"), but GetType returns an int, which also maps to slightly different strings ("Request","Response","Event"). Maybe then we should call it something different and have ErrorResponse as it's own type, so one doesn't need to test for success=true/false.

Reply:
I'm not sure what the proposal here is, can you clarify?

FWIW while working on the Delve PR I realized we don't necessarily need this method at all, so we may just remove it for now and keep only GetSeq. Would that work?

Reply:
Agreed, let's remove it for now.